sycl/include/sycl/handler.hpp

```diff
@@ -1389,7 +1379,8 @@ class __SYCL_EXPORT handler {
     processLaunchProperties(Props);
     setType(detail::CGType::Kernel);
     extractArgsAndReqs();
-    MKernelName = getKernelName();
+    setDeviceKernelInfoPtr(&detail::getDeviceKernelInfo(
+        detail::CompileTimeKernelInfoTy{getKernelName()}));
```

Comment:
In case of `libsycl_preview.so` the `getKernelName()` returns `std::string_view` to the `kernel_impl::MName`. In case of the interop kernel, we might store `std::string_view` to the destructed string.

Reply:
Not part of this PR anymore.